packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.kt

```diff
@@ -169,18 +169,21 @@ open class FileDownloader(context: Context) {
       return
     }
 
-    parseManifest(manifestBodyAndHeaders.first, manifestBodyAndHeaders.second, extensions, configuration, callback)
+    // part headers take precedence over main response headers
+    // for example, content-type should be that of the part (application/json), not multipart/*
+    val combinedHeaders = Headers.Builder().addAll(response.headers()).addAll(manifestPartBodyAndHeaders.second).build()
```

Comment:
For headers that the spec requires to be listed in the body part, I think it would be better to strip them out of the main response headers. For example, if a body part is missing a content-type header, we don't want to imply that the body part's content-type is the same as the response's content-type. Same with content-disposition, which is a valid top-level header (though, in this case if content-disposition were unspecified, we wouldn't get here in the code), and so on.